app/js/lib/view_handler/element_visibility_switcher.js

```diff
@@ -0,0 +1,118 @@
+'use strict';
+
+var objectAssign = require( 'object-assign' ),
+
+	/**
+	 * View Handler for showing and hiding elements if the update value matches a regular expression
+	 * @class
+	 */
+	VisibilitySwitcher = {
+		showOnValueMatchRegex: null,
+		animator: null,
+
+		isHidden: null, // Null to always trigger show/hide on first call to update
+
+		update: function ( value ) {
+			if ( this.showOnValueMatchRegex.test( value ) ) {
+				if ( this.isHidden === false ) {
+					return;
+				}
+				this.animator.showElement();
+				this.isHidden = false;
+			} else {
+				if ( this.isHidden === true ) {
+					return;
+				}
+				this.animator.hideElement();
+				this.isHidden = true;
+			}
+		}
+	},
+
+	ElementSlideAnimator = {
+		el: null,
+
+		// internal fields
+		slideSpeed: 600,
+
+		showElement: function () {
+			this.el
+				.slideDown( this.slideSpeed )
+				.animate(
+					{ opacity: 1 },
+					{ queue: false, duration: this.slideSpeed }
+				);
+		},
+		hideElement: function () {
+			this.el
+				.slideUp( this.slideSpeed )
+				.animate(
+					{ opacity: 0 },
+					{ queue: false, duration: this.slideSpeed }
+				);
+		}
+	},
+
+	ElementHideAnimator = {
```

Comment:
Not sure about the name here. It took me a bit to realize that this class is meant to be of the same type as `ElementSlideAnimator`. That might be somewhat clear if you look at the methods they have, though I started looking at their usage in the factory functions. Both of these do show/hide animations. How about `SlidingElementAnimator` (or `SlideElementAnimator`) and something like `SimpleElementAnimator`? Having the same suffix also makes it more clear they are of the same type. `ElementAnimator` is a lot more general than what these things actually do, so the name is still not very good.